src/components/strategies/common/d3Chart/D3PriceHistory.tsx

```diff
@@ -211,34 +226,61 @@ export const D3PriceHistory: FC<Props> = (props) => {
     domainTolerance: 0.1,
   });
 
-  const invertX = scaleBandInvert(xScale);
-
-  const start = invertX(xScale.bandwidth() / 2) ?? xScale.domain()[0];
-  const end = invertX(dms.boundedWidth) ?? xScale.domain().at(-1);
-
   const disabledDates = [
     {
       before: new Date(Number(xScale.domain()[0]) * 1000),
       after: new Date(Number(xScale.domain().at(-1)!) * 1000),
     },
   ];
 
-  useEffect(() => {
-    const id = setTimeout(() => {
-      if (onRangeUpdates) onRangeUpdates({ start, end });
-    }, 100);
-    return () => clearTimeout(id);
-  }, [onRangeUpdates, start, end]);
-
-  const zoomFromTo = ({ start, end }: { start?: Date; end?: Date }) => {
+  const zoomFromTo = async ({ start, end }: { start?: Date; end?: Date }) => {
     if (!start || !end) return;
-    zoomRange(toUnixUTC(startOfDay(start)), differenceInDays(end, start) + 1);
+    setListenOnZoom(false);
+    const from = toUnixUTC(startOfDay(start));
+    const to = toUnixUTC(startOfDay(end));
+    await zoomRange(from, differenceInDays(end, start) + 1);
+    onRangeUpdates({ start: from, end: to });
+    setListenOnZoom(true);
```

Comment:
If you want to see the listenOnZoom effect immediately, wouldn't a useRef be better here? The two set states will overwrite eachother to only do `setListenOnZoom(true)`

Reply:
Not really because there is a `await` in the middle so React would have the time to rerender. The goal is to force the other useEffect to stop listening on zoom end event.